src/networking.c

```diff
@@ -4215,7 +4216,9 @@ int handleClientsWithPendingWritesUsingThreads(void) {
             installClientWriteHandler(c);
         }
     }
-    listEmpty(server.clients_pending_write);
+    while(listLength(server.clients_pending_write) > 0) {
```

Comment:
It expose the details of releasing a list, and it involve another question, whether it's just ad-hoc optimization, or it's a beginning, more and more list will be substituted with this new list(a kind of intrusive list), as the api changes(listDelNode-> listUnlinkNode, listDelNode->listUnlinkNode and so on), I think mixing them up tend to get misused, maybe introducing a new type is more appropriate.

Reply:
@uvletter I agree that this is basically making a hybrid of intrusive/non-intrusive container, but I think this API can also evolve to fit other use cases (like moving nodes between lists without reallocating memory).
I suppose the main concern is memory leak (and double free which is easier to catch), but maybe we can find a different solution for these (like relying on address sanitizer?)

Reply:
Thanks for the explanation!  I agree that it's legitimate API, and can evolve to support others use cases easily. My worry is if the intrusive list get widely used, but some API are implementation-special, when I wanna del a node, I need firstly distinguish it's intrusive list or not, to determine to use listDelNode or listUnlinkNode. But if it's a type, compiler will check for us, no need for my intelligence or Asan. 

Another point is if new type is used we can even eliminate the `void* val` in `struct listNode`, which save a pointer for every node.

If intrusive list is intended to be used in limited a few performance-critical places, I think the current way is also OK, I'm not meant against it : )

Reply:
Thanks! I appreciate the input